modules/p2/ting_openlist/include/ting_openlist_analysis.inc

```diff
@@ -0,0 +1,150 @@
+<?php
+
+/**
+ * @file
+ * Functions for analysis of local OpenList Entities
+ */
+
+/**
+ * Get providers.
+ *
+ * @return int
+ *   Number of providers.
+ */
+function ting_openlist_local_analysis_data_providers() {
+  $query = db_select('users', 'u');
+  $query->join('users_roles', 'ur', 'ur.uid = u.uid');
+  $query->join('role', 'r', "r.rid = ur.rid AND r.name = 'provider'");
+  return $query->countQuery()->execute()->fetchField();
+}
+
+/**
+ * Get providers with an openlist uid.
+ *
+ * @return int
+ *   Number of providers.
+ */
+function ting_openlist_local_analysis_data_providers_id() {
+  $query = db_select('users', 'u');
+  $query->join('users_roles', 'ur', 'ur.uid = u.uid');
+  $query->join('role', 'r', "r.rid = ur.rid AND r.name = 'provider'");
+  $query->condition('openlist_uid', NULL, 'IS NOT');
+  return $query->countQuery()->execute()->fetchField();
+}
+
+/**
+ * Get providers which is synced.
+ *
+ * @return int
+ *   Number of synced users.
+ */
+function ting_openlist_local_analysis_data_providers_openlist_is_sync() {
+  $query = db_select('users', 'u');
+  $query->join('users_roles', 'ur', 'ur.uid = u.uid');
+  $query->join('role', 'r', "r.rid = ur.rid AND r.name = 'provider'");
+  $query->condition('openlist_modified', 0, '>');
+  return $query->countQuery()->execute()->fetchField();
+}
+
+/**
+ * Count the types of lists.
+ *
+ * @return array
+ *   An associative array, where the key is the list_type_X (X being number of
+ *   lists) and the key is number of occurences.
+ */
+function ting_openlist_local_analysis_data_list_counts() {
+  $sub_query = db_select('field_data_field_list_type', 'flt');
+  $sub_query->join('eck_ding_type', 'dt', 'dt.id = flt.entity_id');
+  $sub_query->addField('flt', 'field_list_type_value', 'type');
+  $sub_query->addExpression('COUNT(flt.entity_id)', 'cnt');
+  $sub_query
+    ->condition('flt.field_list_type_value', 'user_list', '!=')
+    ->groupBy('dt.uid')
+    ->groupBy('flt.field_list_type_value')
+    ->orderBy('cnt', 'DESC');
+  $query = db_select($sub_query, 's');
+  $query->addExpression("CONCAT(s.type, '__', s.cnt)", 'type');
+  $query->addExpression('COUNT(s.type)', 'cnt');
+  $query
+    ->groupBy('s.type')
+    ->groupBy('s.cnt')
+    ->orderBy('s.type')
+    ->orderBy('s.cnt');
+  return $query->execute()->fetchAllKeyed();
+}
+
+/**
+ * The top 10 duplicate lists.
+ *
+ * @return array
+ *   Keyed array, where they key is the openlist_uid and value the number of
+ *   list duplicates.
```

Comment:
I am afraid do not understand the documentation of this function either. It says that the function returns "duplicate lists" but it seems to return an array where the keys are OpenList *user* ids.

Could it be that the method returned the top 10 *users* ordered by the number of duplicate lists they have?